modules/core/client/components/Board.js

```diff
@@ -0,0 +1,14 @@
+import { Children, cloneElement, useEffect } from 'react';
+import { selectPhoto } from '../services/photos.service';
+
+export default function Board({ name='bokeh', children, onNameChanged }) {
+
+  // update
+  useEffect(() => {
+    onNameChanged({ name: selectPhoto(name) });
+  }, [name]);
+
+  const photo = selectPhoto(name);
+  const style = { backgroundImage: `url("${photo.imageUrl}")` };
+  return Children.map(children, child => cloneElement(child, { style }));
```

Comment:
I wonder whether it would be simpler to just make it output a `<section>` element, instead of manipulating the background image of its children, as in all the uses I can see, it's always on a `<section>` element...

It feels kind of magical for it to modify the children.

I think the logic for it to be a directive before is that in angular you had to choose between an something with a template and controller (which felt a bit heavy), and something that just fiddles with the element (a directive), whereas in react they are equally lightweight (something like that anyway, I've slightly forgotten about angular...)

Reply:
Simplified in fc4f0f4bdf7fa8a47447c9f638ac7c9c74ff6623